### PR TITLE
Explore: Fix time interpolation

### PR DESCRIPTION
**Pull request by ifrost**
**What this PR does / why we need it**:

When Explore is opened directly with `/explore` URL the interpolation of `__from` / `__to` variables doesn't work. This is because in such case `TemplateSrv` never receives `timeRange` that is used for interpolation. 

When a Dashboard page is opened first and then the user navigates to Explore the interpolation works fine, though it may use incorrect time range. This is because the time range applied by [this action](https://github.com/grafana/grafana/blob/main/public/app/features/variables/state/actions.ts#L134) becomes stale, never updated but still used for interpolation.

The fix is to update the current time range each time the range changes in `updateTime` action. Note that the flow is slightly different from Dashboards where the time range gets _refreshed_ in the `TemplateSrv` instance when [time picker changes](https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/components/DashNav/DashNavTimeControls.tsx#L41). 

In Explore we call `updateTime` when a query is run or the date range changes. This also involves re-initializing `TimeSrv` (it had already been working this way), because we use **one singleton** used by **both Explore panes**. This looks like a code smell we need to look at later, though.

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/support-escalations/issues/1459

**Special notes for your reviewer**:

How to test it?
1) Go to Explore
2) Use dev-loki
3) Run following query: `{ from="${__from}", to="${__to}" }`
4) Refresh Explore page
5) Check in the network tab that the acutal query that was passed to the backend has the interpolated time.

### CODE REVIEW

File: packages/grafana-runtime/src/services/templateSrv.ts [hunk 1]
```diff
@@ -22,6 +22,11 @@ export interface TemplateSrv {
    * Checks if a target contains template variables.
    */
   containsTemplate(target?: string): boolean;
+
+  /**
+   * Update the current time range to be used when interpolating __from / __to variables.
+   */
+  updateTimeRange(timeRange: TimeRange): void;
```
Comment (ifrost) [inline packages/grafana-runtime/src/services/templateSrv.ts:29]: Not sure if we can make it part of public interface?

File: public/app/features/explore/spec/interpolation.test.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    const fakeFetch = jest.fn();
+
+    (datasources.loki.query as jest.Mock).mockImplementation((request: DataQueryRequest<LokiQuery>) => {
+      fakeFetch(getTemplateSrv().replace(request.targets[0]!.expr));
+      return makeLogsQueryResponse();
+    });
+
+    await waitForExplore();
+
+    expect(fakeFetch).toBeCalledWith('{ from="1600000000000", to="1700000000000" }');
```
Comment (Elfo404) [inline public/app/features/explore/spec/interpolation.test.tsx:37]: would it make sense to test for more URL scenarios here?
like split mode with different ranges?

File: public/app/features/explore/state/time.test.ts [hunk 3]
```diff
@@ -3,9 +3,41 @@ import { dateTime, LoadingState } from '@grafana/data';
...
+  describe('When time is updated', () => {
+    it('Time service is re-initialized and template service is updated with the new time range', async () => {
+      const { dispatch } = configureStore({
+        ...(createDefaultInitialState() as any),
+      });
+      await dispatch(updateTime({ exploreId: ExploreId.left }));
+      expect(mockTimeSrv.init).toBeCalled();
+      expect(mockTemplateSrv.updateTimeRange).toBeCalledWith(MOCK_TIME_RANGE);
+    });
+  });
```
Comment (Elfo404) [inline public/app/features/explore/state/time.test.ts:39]: not sure if this adds any value compared to what may be more robust and maybe less implementation details specific  tests in `public/app/features/explore/spec/interpolation.test.tsx`

**Review by Elfo404 — APPROVED** [review]
LGTM

**Pull request merged**